test/e2e/node_ip_mac_migration.go

```diff
@@ -519,6 +519,11 @@ spec:
 										return true, nil
 									}
 								}
+								// Due to potential k8s bug described here: https://github.com/ovn-org/ovn-kubernetes/issues/4073
+								// We may need to restart kubelet for the backend pod to update its host networked IP address
+								restartCmd := []string{"docker", "exec", workerNode.Name, "systemctl", "restart", "kubelet"}
```

Comment:
I'm not sure why you say it is not supported by k8s. Do you have some links you can share? I dont think it is required to delete and re-add the nodes. The IP is reported in node status, which is subject to change based on what kubelet reports.

The test changes the kubelet IP by modifying the kubelet conf and restarting kubelet. When this happens, kubelet will not change its IP yet because the node does not have this IP configured on any interface:

Apr 03 13:47:36 ovn-worker kubelet[230053]: E0403 13:47:36.446808  230053 kubelet_node_status.go:701] "Failed to set some node status fields" err="failed to validate nodeIP: node IP: \"172.18.0.250\" not found in the host's network interfaces" node="ovn-worker"

Then the test adds this IP to a linux interface. At that point kubelet does update its IP its reporting to the node. However, it does not update the host network pods ip address, until kubelet is restarted. I think this is a bug, and probably just an oversight. I'll report it as a bug in upstream.

Reply:
I wasn't sure, it was more of a question :D sorry that was not phrased correctly but I did some searching around:

1. https://github.com/kubernetes/kubernetes/issues/105320
2. https://github.com/kubernetes/enhancements/pull/3000
3. many issues around this already u/s
4. https://stackoverflow.com/questions/54942488/how-to-change-the-internal-ip-of-kubernetes-worker-nodes
which means we do allow changing this via kubelet? I cannot find any docs around this in k/k which is why I had a doubt on whether its supported or not.

Reply:
Ah it looks like number 2 that you mentioned. Looks like the effort died. I'll make a reference to this in our github issue.

Reply:
> I'm not sure why you say it is not supported by k8s. Do you have some links you can share? I dont think it is required to delete and re-add the nodes. The IP is reported in node status, which is subject to change based on what kubelet reports.
> 
> The test changes the kubelet IP by modifying the kubelet conf and restarting kubelet. 

ack that makes sense seems like what was described in the stackoverflow article.

> When this happens, kubelet will not change its IP yet because the node does not have this IP configured on any interface:
> 
> Apr 03 13:47:36 ovn-worker kubelet[230053]: E0403 13:47:36.446808 230053 kubelet_node_status.go:701] "Failed to set some node status fields" err="failed to validate nodeIP: node IP: "172.18.0.250" not found in the host's network interfaces" node="ovn-worker"

hmm I see

> 
> Then the test adds this IP to a linux interface. At that point kubelet does update its IP its reporting to the node. However, it does not update the host network pods ip address, until kubelet is restarted. 

aha ok yeah you are right if this is officially supported then this certainly sounds like a bug.

Reply:
I think changing node IPs is _supposed to_ work, but it's well known that some components don't handle it, so the recommended way to change a node's IP is to delete the `Node` and let it be recreated with the new IP.

(Note that this only applies to changing the primary node IP; you're definitely allowed to add and remove other elements from `node.status.addresses`, and you're definitely allowed to convert a node between single- and dual-stack.)